ipv8/messaging/anonymization/community.py

```diff
@@ -354,15 +361,21 @@ def remove_relay(self, circuit_id, additional_info='', destroy=False, got_destro
         if destroy:
             self.destroy_relay(to_remove, got_destroy_from=got_destroy_from)
 
+        removed_relays = []
         for cid in to_remove:
             # Remove the relay
             self.logger.info("Removing relay %d %s", cid, additional_info)
-            relay = self.relay_from_to.pop(cid, None)
+            removed_relays.append(self.relay_from_to.pop(cid, None))
 
             # Remove old session key
             if cid in self.relay_session_keys:
                 del self.relay_session_keys[cid]
 
+            if cid in self.directions:
+                del self.directions[cid]
```

Comment:
This sort of construction is actually rather fragile. We should start using `self.somedict.pop(somekey, None)`.